app/lib/report_helper.rb

```diff
@@ -530,20 +530,22 @@ def self.fetch_top_taxons(samples, background_id, categories, read_specificity =
         pr = pipeline_runs_by_id[pipeline_run_id]
         result_hash[pipeline_run_id] = { "pr" => pr, "taxon_counts" => [] }
       end
-      row["rpm"] = row["r"] / ((pr.total_reads - pr.total_ercc_reads.to_i) * pr.subsample_fraction) * 1_000_000.0
-      row["zscore"] = row["stdev"].nil? ? ZSCORE_WHEN_ABSENT_FROM_BACKGROUND : ((row["rpm"] - row["mean"]) / row["stdev"])
-      row["zscore"] = ZSCORE_MAX if row["zscore"] > ZSCORE_MAX && row["zscore"] != ZSCORE_WHEN_ABSENT_FROM_BACKGROUND
-      row["zcore"] = ZSCORE_MIN if row["zscore"] < ZSCORE_MIN
-      result_hash[pipeline_run_id]["taxon_counts"] << row
+      if pr.total_reads
```

Comment:
Was this in response to a bug, or just adding for best practice?

Reply:
This was in response to an error I saw testing some random samples. It seems like a sensible guard. However, having `nil` total_reads may indicate an unaddressed bug further upstream. I'm not sure if that is a valid state.

Reply:
I think the frontend won't let you select samples whose pipeline run is not successful, so that might have masked issues in the past